PyChakra/__init__.py

```diff
@@ -10,103 +10,230 @@
 
 
 import ctypes
-import platform
-import sys
+import re
 import json
+import threading
 from os import path
+from sys import platform
+
+
+try:
+    unicode
+except NameError:
+    unicode = str
+
+try:
+    bytes
+except NameError:
+    bytes = str
+
+preferredEncoding = None
+
+def getpreferredencoding():
+    global preferredEncoding
+
+    if preferredEncoding is None:
+        import locale
+        preferredEncoding = locale.getpreferredencoding(False)
+
+    return preferredEncoding
 
 
 def get_lib_path():
     root = path.dirname(__file__)
 
-    if sys.platform == "darwin":
+    if platform == "darwin":
         return path.join(root, "libs/osx/libChakraCore.dylib")
 
-    if sys.platform.startswith("linux"):
+    if platform.startswith("linux"):
         return path.join(root, "libs/linux/libChakraCore.so")
 
-    if sys.platform == "win32":
-        if platform.architecture()[0].startswith("64"):
+    if platform == "win32":
+        from platform import architecture
```

Comment:
This breaks pep8 https://www.python.org/dev/peps/pep-0008/#imports

```
Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.```

Reply:
I think it doesn't matter, we don't need to import un-used modules.